src/main/java/io/warp10/ext/interpolation/InterpolatorOnTicks.java

```diff
@@ -0,0 +1,159 @@
+
+    private final PolynomialSplineFunction func;
+    private final String generatedFrom;
+    private GeoTimeSerie gts;
+
+    private MAPPER_INTERPOLATOR(PolynomialSplineFunction function, String interpolatorName, GeoTimeSerie gts) {
+      super(interpolatorName);
+      func = function;
+      this.gts = gts;
+      generatedFrom = interpolatorName;
+    }
+
+    private double value(double x) {
+      if (!func.isValidPoint(x)) {
+        return Double.NaN;
+      } else {
+        return func.value(x);
+      }
+    }
+
+    @Override
+    public Object apply(Object[] args) throws WarpScriptException {
+      long tick = (long) args[0];
+      long[] ticks = (long[]) args[3];
+      long[] locations = (long[]) args[4];
+      long[] elevations = (long[]) args[5];
+      Object[] values = (Object[]) args[6];
+
+      if (1 < ticks.length) {
+        throw new WarpScriptException(getName() + " expects at most 1 tick but got " + ticks.length);
+      }
+
+      //
+      // We return the same value if a value is present,
+      // else, we interpolate.
+      //
+
+      if (1 == values.length) {
+        double value = ((Number) values[0]).doubleValue();
+
+        if (!Double.isNaN(value)) {
+          return new Object[] {tick, locations[0], elevations[0], value};
+
+        } else {
+          return new Object[] {tick, locations[0], elevations[0], value(tick)};
+        }
+      }
+
+      return new Object[] {tick, GeoTimeSerie.NO_LOCATION, GeoTimeSerie.NO_ELEVATION, value(tick)};
+    }
+
+    @Override
+    public String toString() {
+      StringBuilder sb = new StringBuilder();
+      try {
+        SNAPSHOT.addElement(sb, gts);
+        sb.append(" ");
+      } catch (Exception e) {
+        throw new RuntimeException("Error building argument snapshot", e);
+      }
+
+      sb.append(generatedFrom);
+
+      return sb.toString();
+    }
+  }
+
+  public InterpolatorOnTicks(String name, TYPE type) {
+    super(name);
+    this.type=type;
+  }
+
+  @Override
+  public Object apply(WarpScriptStack stack) throws WarpScriptException {
+
+    double xval[];
+    double fval[];
+
+    Object o = stack.pop();
+
+    if (!(o instanceof GeoTimeSerie)) {
+      throw new WarpScriptException(getName() + " expects a GTS");
+    }
+    GeoTimeSerie gts = (GeoTimeSerie) o;
+    
+    // ticks need to be sorted in order to create the interpolating function
+    GTSHelper.sort(gts);
+
+    // fill x and f
+    int size = gts.size();
+    if (getName().equals("AKIMASPLINEFIT") && size < 5) {
+      throw new WarpScriptException(getName() + " expects at least 5 interpolation points");
+    }
+    xval = new double[size];
+    fval = new double[size];
+    for (int i = 0; i < size; i++) {
+      xval[i] = ((Number) GTSHelper.tickAtIndex(gts, i)).doubleValue();
+      fval[i] = ((Number) GTSHelper.valueAtIndex(gts, i)).doubleValue();
+    }
```

Comment:
check gts type before extracting values, or it will fail on boolean or string type.

As it is, there is a non obvious cast error.